.github/workflows/ci.yml

```diff
@@ -330,6 +330,42 @@ jobs:
       - run: python -m pytest modin/test/test_logging.py
       - uses: codecov/codecov-action@v2
 
+  test-no-engine:
```

Comment:
Why do we need to add this new job and `requirements/requirements-no-engine.yml` if we already test `test_dispatcher.py` file? `test_add_option` test in the file is in charge of that you are going to add.

Reply:
To reproduce #5308 , I needed an environment without any default engine. Without a test like this, we won't catch a regression back to #5308 . Please let me know if there's a better way.

Reply:
Ok, let's leave as is for now.